looplib/looptools.py

```diff
@@ -204,3 +208,85 @@ def FRiP(number_of_sites, lef_positions, boundary_positions):
     return np.sum(hist[boundary_positions]) / len(lef_positions)
 
 
+def calc_coverage(LEF_array, L):
+    """
+    Calculate the average coverage (fraction of polymer covered by at least one loop)
+    
+    Args:
+        LEF_array (ndarray): Nx2 numpy array of extruder positions
+        L (int): Total polymer length.
+
+    Returns:
+        float: Loop coverage.
+    """
+    
+    coverage = np.zeros(int(L))
+    
+    for p in LEF_array:
+        coverage[p[0]:p[1]+1] = 1
+        
+    return coverage.sum() / float(L)
+
+
+def _get_collided_pairs(LEF_array, r):
```

Comment:
is `r` best as a float as indicated or an int? 

and should the r=1 default be here as well?

Reply:
Slightly unsure how KDTree handles strict inequalities in this case, since it converts position data to double's under the hood. So it's perhaps safer to keep r as a float, and use a small buffer for sanity (e.g., set r=1.01 to get all adjacent extruders without worrying about potential rounding errors)

Reply:
Gotcha-- if we go with that perhaps a comment in the docstring and/or some TOL=0.1 (added in later) global variable might be helpful?